packages/xrpl/src/Wallet/index.ts

```diff
@@ -434,15 +443,20 @@ export class Wallet {
    *
    * @param signedTransaction - A signed transaction (hex string of signTransaction result) to be verified offline.
    * @returns Returns true if a signedTransaction is valid.
+   * @throws {Error} Transaction is missing a signature, TxnSignature
    */
   public verifyTransaction(signedTransaction: Transaction | string): boolean {
     const tx =
       typeof signedTransaction === 'string'
         ? decode(signedTransaction)
         : signedTransaction
     const messageHex: string = encodeForSigning(tx)
-    const signature = tx.TxnSignature
-    return verify(messageHex, signature, this.publicKey)
+    // Need a SignedTransaction class where TxnSignature is not optional.
+    if (typeof tx.TxnSignature !== 'string') {
```

Comment:
This should also check if the `TxnSignature` is an empty string

Reply:
Or does that not matter because `verify` won't work?

Reply:
This is largely to keep TypeScript happy. 

What happened here is that verify() used to be `any` typed for the signature parameter, and during the move to @noble the typing was tightened up. That exposed this situation here where tx.TxnSignature can be string|undefined. In the case where the signature is an empty string, verify will definitely catch it (as it did before )

Reply:
I combined aspects of `Wallet.verifyTransaction` and `verifySignature` and added an additional test.